vernac/vernacentries.ml

```diff
@@ -754,6 +754,22 @@ let vernac_assumption ~atts discharge kind l nl =
     Attributes.unsupported_attributes [CAst.make ("using",VernacFlagEmpty)];
   ComAssumption.do_assumptions ~poly:atts.polymorphic ~program_mode:atts.program ~scope ~kind ?deprecation:atts.deprecated nl l
 
+let vernac_context ~atts (sup : Constrexpr.local_binder_expr list) =
+  if Dumpglob.dump () then begin
+    let sec = Lib.sections_are_opened () in
+    let ty = if sec then "var "else "ax" in
+    let l = List.map (function
+                | Constrexpr.CLocalAssum (l, _, _, _) -> l
+                | Constrexpr.CLocalDef (n, _, _, _) -> [n]
+                | Constrexpr.CLocalPattern _ -> [])
+              sup in
+    List.iter (function
+        | {v = Names.Name.Anonymous; _} -> ()
+        | {v = Names.Name.Name id; loc} ->
+           Dumpglob.dump_definition (CAst.make ?loc id) sec ty)
+      (List.flatten l) end;
+  ComAssumption.do_context ~poly:(only_polymorphism atts) sup
```

Comment:
Why isn't the dumping in do_context?

Reply:
Most of the dumping seems to happen in `vernacentries.ml` but you're probably right that, while at it it should be done in `comAssumption.ml`